app/Model/Server.php

```diff
@@ -4327,30 +4327,57 @@ private function __attachRecoveryQuery($field, $table)
                 switch($field['error_type']) {
                     case 'missing_column':
                         $field['sql'] = sprintf(
-                            'ALTER TABLE `%s` ADD COLUMN `%s` %s%s %s %s;',
+                            'ALTER TABLE `%s` ADD COLUMN `%s` %s%s %s %s %s;',
                             $table,
                             $field['column_name'],
                             $field['expected']['data_type'],
                             $length !== null ? sprintf('(%d)', $length) : '',
-                            isset($field['expected']['default']) ? 'DEFAULT "' . $field['expected']['default'] . '"' : '',
+                            isset($field['expected']['column_default']) ? $field['expected']['column_default'] . '"' : '',
                             $field['expected']['is_nullable'] === 'NO' ? 'NOT NULL' : 'NULL',
                             empty($field['expected']['collation_name']) ? '' : 'COLLATE ' . $field['expected']['collation_name']
                         );
                         break;
                     case 'column_different':
                         $field['sql'] = sprintf(
-                            'ALTER TABLE `%s` CHANGE `%s` `%s` %s%s %s %s;',
+                            'ALTER TABLE `%s` MODIFY COLUMN `%s` %s%s %s %s %s;',
                             $table,
                             $field['column_name'],
-                            $field['column_name'],
                             $field['expected']['data_type'],
                             $length !== null ? sprintf('(%d)', $length) : '',
-                            isset($field['expected']['default']) ? 'DEFAULT "' . $field['expected']['default'] . '"' : '',
+                            isset($field['expected']['column_default']) ? 'DEFAULT "' . $field['expected']['column_default'] . '"' : '',
                             $field['expected']['is_nullable'] === 'NO' ? 'NOT NULL' : 'NULL',
                             empty($field['expected']['collation_name']) ? '' : 'COLLATE ' . $field['expected']['collation_name']
                         );
                         break;
                 }
+            } elseif($field['error_type'] == 'missing_table') {
```

Comment:
Not convinced by this. We currently basically only correctly configure types that we know in advanced - such as the type based length setting. If a table has anything we do not expect that script will create inaccurate settings. Perhaps what we should consider is loading new tables from MYSQL.sql directly and being stricter on keeping it up to date?

Reply:
In that case, it means that we expect the schema to have a table which does not exists in the diagnosed database. As we know the complete configuration of the missing table thanks to the expected schema, we can also provide a quick fix.
I'm not sure if this issue will ever happen but if it does, users can quick fix it.

Reply:
"As we know the complete configuration of the missing table thanks to the expected schema"

We do, but the current builder doesn't know how to correctly interpret it. If we build something more complete - sure.

Reply:
Indeed, I'm not considering indexes. However straight of my mind, I don't see how it's different from a column addition. The column configuration are basically the same.
In what sense could I improve it?
Although, I don't see a situation where we would end up missing a table so I can also scrap up this part of the code.

Reply:
Indexes are one thing, I'm more thinking types. The part above will for example set the lengths based on type, but in some cases might set it incorrectly. In those cases we simply don't provide a recovery script for column level issues, but omitting a column in a table add is an issue. However, in those cases we could just grab the full table definition from mysql.sql

Reply:
Still provide the SQL query but added a warning message